File: Python/maximum-difference-between-adjacent-elements-in-a-circular-array.py
# Time:  O(n)
# Space: O(1)

# array
class Solution(object):
    def maxAdjacentDistance(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        return max(abs(nums[i]-nums[i-1]) for i in xrange(len(nums)))

File: C++/maximum-difference-between-adjacent-elements-in-a-circular-array.cpp
// Time:  O(n)
// Space: O(1)

// array
class Solution {
public:
    int maxAdjacentDistance(vector<int>& nums) {
        int result = 0;
        for (int i = 0; i < size(nums); ++i) {
            result = max(result, abs(nums[i] - (i - 1 >= 0 ? nums[i - 1] : nums.back())));
        }
        return result;
    }
};
